app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CategoriesController < ApplicatinController
+    before_action :set_task, only: [:update, :destroy]
+    before_action :set_new_task, only: [:list]
+    before_action :set_search, only: [:list]
+
+    def list
+        @category = @search.result.on_user.sort_by(:order_row)
+    end
+
+    def set_category
+        @category = Category.on_user.find_by(id: params[:id])
+    end
+
+    def new
+        @category = Category.new
```

Comment:
__[rubocop]__
* Style/IndentationWidth: Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)